year18/src/day11.rs

```diff
@@ -1,10 +1,54 @@
 // Potential improvements:
 //
+use std::i32::MIN;
+use nalgebra::DMatrix;
 
-pub fn day11(_input_lines: &[Vec<String>]) -> (String, String) {
-    let answer1 = 0;
-    let answer2 = 0;
-    (format!("{}", answer1), format!("{}", answer2))
+const POWER_GRID_SIZE: usize = 300;
+
+pub fn day11(input_lines: &[Vec<String>]) -> (String, String) {
+    let serial_number = input_lines[0][0].parse::<i32>().unwrap();
+
+    // The question is 1-indexed, but the matrix is 0-indexed.
+    let power_grid = DMatrix::<i32>::from_fn(
+        POWER_GRID_SIZE, 
+        POWER_GRID_SIZE, 
+        |y, x| ((x as i32 + 11) * (y as i32 + 1) + serial_number) * (x as i32 + 11) / 100 % 10 - 5
```

Comment:
brackets can be useful in expressions like these as it's not immediately obvious if this will be interpreted as `(a / 100) % 10` or `a / (100 % 10)`